rust/src/worker.rs

```diff
@@ -624,13 +624,28 @@ impl Worker {
     }
 
     /// Updates the worker settings in runtime. Just a subset of the worker settings can be updated.
-    pub async fn update_settings(&self, data: WorkerUpdateSettings) -> Result<(), RequestError> {
+    pub async fn update_settings(&self, _data: WorkerUpdateSettings) -> Result<(), RequestError> {
```

Comment:
@nazar-pc, first request sent with body.

Request body is created out of `WorkerUpdateSettings` (not in this case yet, we're just sending a default one for now) with the FBS builder in `channel`. Then `channel.request_fbs()` is called given the method and body and the response is converted accordingly.

There are two concerns here:
1. The same FBS builder must be used to create the request body and to serialize the request later by `channel.request_fbs()`. No other `builder` usage should exist within those. IMHO current implementation is not thread safe on this regard, since another thread can use the builder between this method and `channel.request_fbs()`.
2.  I'm not sure if this flow (FBS body creation, calling to `channel.request_fbs()` and response conversion can be automated with macros).

Reply:
1. What I mean is that the builder is used as follows:

a) Create the body of the request (in Worker, Transport, Producer, Consumer)
b) In `request_fbs()` in `channel.rs` where the whole request is serialized.

My concern is that between `a)` and `b)` for a given request another thread could attempt to send another request and we would get the builder buffer corrupted. Ie:

* thread1 `worker::update_settings`: create the request body.
* thread2 `transport::consume`: create the request body (oh.., the builder buffer is now polluted).
* thread1 `channel:request_fbs`: the builder buffer polluted.

In order to avoid that, a builder is now created for each request and passed to `channel:request_fbs`. As an optimization we could use directly the memory of the builder, since we own it, rather than creating a copy. That's a TODO.

2. :+1: